modules/javafx.controls/src/main/java/javafx/scene/control/ListCell.java

```diff
@@ -540,24 +541,27 @@ private void updateEditing() {
         final int editIndex = list == null ? -1 : list.getEditingIndex();
         final boolean editing = isEditing();
 
-        // Check that the list is specified, and my index is not -1
-        if (index != -1 && list != null) {
-            // If my index is the index being edited and I'm not currently in
-            // the edit mode, then I need to enter the edit mode
-            if (index == editIndex && !editing) {
-                startEdit();
-            } else if (index != editIndex && editing) {
-                // If my index is not the one being edited then I need to cancel
-                // the edit. The tricky thing here is that as part of this call
-                // I cannot end up calling list.edit(-1) the way that the standard
-                // cancelEdit method would do. Yet, I need to call cancelEdit
-                // so that subclasses which override cancelEdit can execute. So,
-                // I have to use a kind of hacky flag workaround.
+
+        if (editing && (index == -1 || list == null || index != editIndex)) {
+            // If my index is not the one being edited then I need to cancel
```

Comment:
probably just me not being able to see at a glance if this if the overall logic in the method is the exact same as before the fix (modulo the fix itself :), but: I would prefer an extra method used in both early return for the -1/null-list and the old else-if. 

If the second reviewer sees the equivalence at a glance, I'll believe his eyes :)

Reply:
I've simplified the if-else statements. Should now be obvious that it's the same as in TreeTableCell.